README.md

```diff
@@ -0,0 +1,206 @@
+
+## 🛠 Built With <a name="built-with"></a>
+
+### Tech Stack <a name="tech-stack"></a>
+
+<details>
+  <summary>Client</summary>
+  <ul>
+    <li><a href="https://www.w3schools.com/html/default.asp">HTML</a></li>
+    <li><a href="https://www.w3schools.com/css/default.asp">CSS</a></li>
+    <li><a href="https://getbootstrap.com/">Bootstrap</a></li>
+    <li><a href="https://react.dev/learn">React</a></li>
+  </ul>
+</details>
+
+<details>
+  <summary>Server</summary>
+  <ul>
+    <li><a href="https://www.ruby-lang.org/en/">Ruby</a></li>
+    <li><a href="https://guides.rubyonrails.org/">Ruby on Rails</a></li>
+  </ul>
+</details>
+
+<details>
+<summary>Database</summary>
+  <ul>
+    <li><a href="https://www.postgresql.org/">PostgreSQL</a></li>
+  </ul>
+</details>
+
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+- **[User authentication through username]**
+- **[Navigation panel with links to classes]**
+- **[Reservation form]**
+- **[My reservations page.]**
+- **[Add class and Delete class options]**
+- **[Display a list of selected classes on the main page]**
+- **[Detailed class view with a "Reserve" button]**
+- **[Responsive design for both mobile and desktop]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [EduPlanner](https://edu-planner.onrender.com)
+
+## 🔗 Back End Link <a name="documentation"></a>
+
+- [EduPlanner Backend](https://github.com/ab-noori/EduPlanner-Backend)
+
+  > _Our backend code is stored in a separate repository and is built using ruby on rails. The backend is responsible for processing data and handling API requests from the frontend. To access our backend code, please click on the provided link below._
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- GETTING STARTED -->
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+To get a local copy up and running, follow these steps.
+
+### Prerequisites
+
+In order to run this project you need:
+
+- [Ruby on Rails](https://rubyonrails.org/) installed and running. To get more information, read the [installation guide](https://guides.rubyonrails.org/).
+
+- [PostgreSQL](https://www.postgresql.org/) installed and running
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+##
+
+### Setup
+
+Clone this repository to your desired folder:
+
+```sh
+  cd my-folder
+  https://github.com/ab-noori/EduPlanner-Frontend.git
+```
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+### Install
+
+Install this project with:
+
+```sh
+    cd EduPlanner-Frontend
+    npm install
+```
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+### Usage
+
+To run the project, execute the following command:
+
+```sh
+  npm start
```

Comment:
- [ ] OPTIONAL: At the time of this review, after spending some time trying to run the project correctly without CORS blocking the local frontend, I noticed the project has been set to only be tested in the live demo, and not in the local dev environment.

This is because the local frontend project uses `https://edu-planner.onrender.com/` for all requests and the deployed version of CORS will only allow that exact origin for the requests, and nothing else:

![image](https://github.com/ab-noori/EduPlanner-Frontend/assets/57421823/d71737ff-0dcb-4c51-94a1-36c08a7a57d2)

I tested accessing with localhost:3000, localhost:3001, http://127.0.0.1:3000/, http://127.0.0.1:3001/.


In order to make your app capable of being tested locally (the frontend), you could add another allowed origin to the deployed version of CORS like this:

```ruby

# how to allow multiple allowed origins in cors.rb, by using an "allow" block for each origin:

Rails.application.config.middleware.insert_before 0, Rack::Cors do
  allowed_headers = %i(get post put patch delete options head)
  allow do
    origins 'https://edu-planner.onrender.com/'
    resource '*', 
    headers: :any, 
    methods: allowed_headers
  end

  allow do
    origins 'http://localhost:3000'
    resource '*', 
    headers: :any, 
    methods: allowed_headers
  end
end
```

This is explained [here](https://github.com/cyu/rack-cors/issues/178).

That way, you can have CORS allow `https://edu-planner.onrender.com/`, and also localhost:3000, 3001, etc.

I suggest adding the allowed origin, and/or the pertinent notes to your readme, in order to increase your chances in the final evaluation as it is customary to clone the projects and review them locally.  

As an additional note, you could also add to your readme instructions about how to correctly set the project to be run locally.

That way, confusion will be prevented and your users (e.g. the final evaluator and potential employers) will be able to easily test your project. 👍